readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/writer/ProtoParquetWriterWithOffset.java

```diff
@@ -63,9 +64,18 @@ public Path close() throws IOException {
         writer.close();
 
         final Path finalPath = new Path(finalHdfsDir, fileNamer.computePath(dayStartTime, latestOffset));
-        fs.rename(temporaryHdfsPath, finalPath);
 
-        LOGGER.info("Committed " + finalPath.toUri());
+        if (!FileSystemUtils.ensureDirectoriesExist(Collections.singleton(finalPath.getParent()), fs)) {
+            LOGGER.warn("Couldn't ensure {} exists", finalPath.getParent());
+            return null;
```

Comment:
I would prefer if we can avoid returning null. I have not looked through the entire code so I don't know if we use the return value from close() method aside from tests. If not, we could return void and do testing based on behaviour (using verify) to make sure we existed the method when wanted.